README.md

```diff
@@ -1,2 +1,359 @@
+external networks in OVN. Configuring access to those VMs is based on:
+* Ports
+* Protocols
+* Remote IPs
+* group membership
+* ether type
+...
+
+In the networking API, defined in OpenStack by Neutron, there is an entity to
+enable these use cases, through [security groups](https://developer.openstack.org/api-ref/network/v2/#security-groups-security-groups).
+Using security groups, the user could - for instance - configure ICMP access to
+all the VMs in OVN (or a subset of VMs), limit the traffic in the OVN external
+networks to web traffic (e.g. tcp port 80). More complex scenarios can be
+achieved, for instance, scenarios where some VMs can access a set of services,
+whereas others cannot. This latter scenario can be configured either by IP,
+subnet CIDR, or semantically, through group membership.
+
+In oVirt, this networking API concept was implement in the
+[ovirt-provider-ovn](https://github.com/oVirt/ovirt-provider-ovn) project, and
+a thorough design document for that feature is found in the following
+[feature page ](https://ovirt.org/develop/release-management/features/network/networking-api-security-groups.html).
+
+The goal of this repo is to provide tools to help the configuration of
+security groups, since currently there is no official supported way to do so
+other than through the provider's RESTful interface. Furthermore, some example
+configurations are also provided.
+
+## Provided tools
+This repo adds tools to help manage the security groups in oVirt, since
+currently there is no supported mechanism to handle those.
+
+This repo provides the following tools:
+* list_openstack_entities.py: python script that list the logical ports or
+  the security groups found in the system. It is very useful to get the IDs
+  of the desired security group, and/or logical port.
+
+  **NOTE:** the openstack CLI could also be used.
+* update_port_security ansible playbook: this playbook's goal is to update
+  port security for the target entities - whether they are networks, or ports.
+* update_ports ansible playbook: this playbook's goal is to update the port's
+  security group membership. Can clear the security groups of a logical port.
+* create_icmp_security_group: this playbook creates (or deletes) a security
+  group and provisions it with a rule allowing ICMP traffic.
+
+  **NOTE:** this playbook provides an example on how the user can configure
+  rules enabling access based on protocol - e.g. to all VMs.
+* create_web_based_security_group.yml: this playbook creates (or deletes) a
+  group that creates the security groups - and rules - to set up a web based
+  scenario.
+
+  **NOTE:** this playbook is meant as an example of how the user can configure
+  rules enabling access to a subset of VMs - in this example, though group
+  membership.
+
+## Requirements
+* python
+* ansible
+* openstacksdk
+
+To properly install openstacksdk, execute:
+
+```bash
+./install_dependencies.sh
+```
+
```

Comment:
The target audience for this document is well educated, means that such a script is nice, but some lines for cop-and-paste would be fine, too.
Would you script work on plain RHEL or CentOS?

Reply:
Right, it's really stupid having a script w/ 1 line of code. When I started it, I thought it would be longer -_-

No clue about RHEL / centos, but, this is PIP we're talking about - it will pull the same code everywhere, so I'm guessing (with a high degree of confidence) yes.

I can add a note inidicating this was only tested on fedora (I really don't plan on checking if this works for other OSs, I've spent plenty of time on this already).

Reply:
The ugly thing is that pip is not part of RHEL/CentOS, so you have to install pip from epel repo
Your target audience is potentially using RHEL.

Reply:
This code is not meant to be used on the engine VM.

Never the less, it can always be executed in a docker container running elsewhere (even if on RHEL land). I could actually provide a docker file to this repo that would set it up.

Reply:
>  I could actually provide a docker file to this repo that would set it up.
Don't put too much effort in, if you just write the command lines for copy and past for RHEL, this should be enough. I expect this would be less than 5 lines (enable epel, install pip, install openstacksdk via pip)

Reply:
Done